public/app/features/variables/state/actions.ts

```diff
@@ -563,7 +565,7 @@ const createGraph = (variables: VariableModel[]) => {
       }
 
       if (variableAdapters.get(v1.type).dependsOn(v1, v2)) {
-        g.link(v1.name, v2.name);
+        g.link(v1.id, v2.id);
```

Comment:
Changed this to match the identifier we're tracking in the `visitedVariables` array since we get nodes from the graph using data from that array. The reason `id` is used instead of name in `visitedVariables` is b/c `KeyedVariableIdentifier` doesn't have the variable name. I think using `id` is sufficient here but the extra data in `KeyedVariableIdentifier` (type and rootStateKey) made me wonder if perhaps the entire object ought to be serialized into a string and then we use that instead of just `id` for node names.